## Division

In addition to the normal divison (`/`, called the true division operator), python has other operator called the integer divison operator (`//`). 

### True division operator 

Returns the result of normal division, which could be a faction (floating point):

In [26]:
7/4

1.75


### Integer division operator `//`

The interger division operator *floored* the result of true division operator:

In [27]:
7 // 4

1

Recall that the `flooring` always rounds toward minus infinity: 

In [28]:
- 7 // 4

-2

In the above, the true divison of `-7` by `4` is `-1.75`. The two nearest integer of `-1.75` is `-1` and `-2`. So the result will be `-2` because it is closer to the minus infinity.
 
This is contrast to the buil-in `int()` function, which simply truncate the faction:

In [29]:
int(-7/4)

-1